### DISCUSSION TITLE
DII from DHQ3?

### DISCUSSION THREAD

**Discussion by wolffm24** [jamesjiadazhan/dietaryindex#291] [Q&A] [answered]
Hi All! I am collecting dietary data for my dissertation in a cohort of people using the DHQ3. I am going to calculate the DII from the DHQ3 data for analysis with outcomes. Is this possible via this package?

Thanks in advance :)
Morgan

**Comment by jamesjiadazhan** [comment, marked as answer]
Hi, thanks for using dietaryindex!

Currently, it is not available in the package. There are two options: you can either use the general function, DII(), by processing the DHQ3 data by yourself so that each data input is matched to the DII () requirements, or you can send me an email and we can work on this together (authorship is expected, as the work is not trivial).

**Reply by wolffm24** [reply]
I would love to work on this together! My school email is mowolff@uiowa.edu
I have begun to work on the code in SAS but can switch to R now.

Morgan

**Reply by wolffm24** [reply]
Hi again James,

Wondering how easy it would be to include energy adjusted DII (E-DII) in the code as well?

Best,
Morgan

**Reply by jamesjiadazhan** [reply]
This is typically done by users. Below is my NHANES example. You don't have to use svyglm for studies that are not survey design. Just use glm.
```

# create a function to adjust a variable for total energy intake
adjust_for_energy <- function(var){
  # create a formula for the model
  formula <- as.formula(paste(var, "~ TOTALKCAL_PHDI"))
  # fit the model
  model <- svyglm(formula, PHDI_GHG_FINAL_DATA_0518_design)
  # predict the variable
  predicted <- predict(model, type = "response")
  # return the predicted variable
  return(predicted)
}

# adjust all component scores for total energy intake
PHDI_GHG_FINAL_DATA_0518$PHDI_WGRAIN_kcal_adjusted <- adjust_for_energy("PHDI_WGRAIN")
```

**Reply by wolffm24** [reply]
Amazing, thank you so much!
